custom_components/petlibro/button.py

```diff
@@ -0,0 +1,74 @@
+"""Support for PETLIBRO buttons."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from functools import cached_property
+from typing import Any, Generic
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import PetLibroHubConfigEntry
+from .entity import PetLibroEntity, _DeviceT, PetLibroEntityDescription
+from .devices.device import Device
+from .devices.feeders.feeder import Feeder
+
+@dataclass(frozen=True)
+class RequiredKeysMixin(Generic[_DeviceT]):
+    """A class that describes devices button entity required keys."""
+
+    set_fn: Callable[[_DeviceT], Coroutine[Any, Any, None]]
+
+@dataclass(frozen=True)
+class PetLibroButtonEntityDescription(ButtonEntityDescription, PetLibroEntityDescription[_DeviceT], RequiredKeysMixin[_DeviceT]):
+    """A class that describes device button entities."""
+
+    entity_category: EntityCategory = EntityCategory.CONFIG
+
+
+DEVICE_BUTTON_MAP: dict[type[Device], list[PetLibroButtonEntityDescription]] = {
+    Feeder: [
+        PetLibroButtonEntityDescription[Feeder](
+            key="manual_feed",
+            translation_key="manual_feed",
+            set_fn=lambda device: device.set_manual_feed()
+        )
+    ]
+}
+
+class PetLibroButtonEntity(PetLibroEntity[_DeviceT], ButtonEntity):
+    """PETLIBRO button entity."""
+
+    entity_description: PetLibroButtonEntityDescription[_DeviceT]
```

Comment:
This makes the voices stop

```suggestion
class PetLibroButtonEntity(PetLibroEntity[_DeviceT], ButtonEntity):  # type: ignore [reportIncompatibleVariableOverride]
    """PETLIBRO button entity."""

    entity_description: PetLibroButtonEntityDescription[_DeviceT]  # type: ignore [reportIncompatibleVariableOverride]
```